api/client/grpc.go

```diff
@@ -0,0 +1,107 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"time"
+
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/celestiaorg/celestia-node/libs/utils"
+)
+
+// GRPCConfig combines all configuration fields for managing the relationship with a Core node.
+type GRPCConfig struct {
```

Comment:
I would avoid calling this just "gRPC," since that's a transport protocol and not the target. Or rather, instead of using GRPC and Core interchangeably, Core should use GRPC under the hood, and never expose this detail.

We may have other stuff using gRPC in the future. Same with RPC.